packages/@aws-cdk/aws-eks/lib/helm-chart.ts

```diff
@@ -0,0 +1,116 @@
+import { CustomResource, CustomResourceProvider } from '@aws-cdk/aws-cloudformation';
+import * as lambda from '@aws-cdk/aws-lambda';
+import { Construct, Duration, Stack } from '@aws-cdk/core';
+import * as path from 'path';
+import { Cluster } from './cluster';
+import { KubectlLayer } from './kubectl-layer';
+
+/**
+ * Helm Chart Properties.
+ */
+export interface HelmChartProps {
```

Comment:
Extract all properties besides `cluster` to `HelmChartOptions` and use it as a base interface for `HelmChartProps`. 

Then, change the signature of `cluster.addChart` to:

```ts
addChart(id: string, options: HelmChartOptions): HelmChart
```

This is the pattern we normally use for cases like this.

Reply:
Ok